src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -779,32 +779,84 @@ Compute the temperature that is consistent with
 See also [`saturation_adjustment`](@ref).
 """
 function saturation_adjustment_q_tot_θ_liq_ice(θ_liq_ice::FT, q_tot::FT, ρ::FT, p::FT) where {FT<:Real}
-  T_1 = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, p) # Assume all vapor
+  T_1 = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, p, PhasePartition(q_tot)) # Assume all vapor
   q_v_sat = q_vap_saturation(T_1, ρ)
   if q_tot <= q_v_sat # If not saturated
     return T_1
   else  # If saturated, iterate
     T_2 = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, p, PhasePartition(q_tot, FT(0), q_tot)) # Assume all ice
     T, converged = find_zero(
-      T -> θ_liq_ice - liquid_ice_pottemp_sat(T, p, PhasePartition_equil(T, ρ, q_tot)),
+      T -> θ_liq_ice - liquid_ice_pottemp_sat_given_pressure(T, p, PhasePartition_equil(T, ρ, q_tot)),
```

Comment:
I'm confused here that `liquid_ice_pottemp_sat_given_pressure` takes pressure `p` as input argument, but `PhasePartition_equil` uses density `rho`. How is consistency ensured (`p = \rho * R * T`)?